src/www/app/outline_server_repository/index.ts

```diff
@@ -144,13 +145,39 @@ export class OutlineServerRepository implements ServerRepository {
   }
 
   validateAccessKey(accessKey: string) {
-    const alreadyAddedServer = this.serverFromAccessKey(accessKey);
+    if (!this.isDynamicAccessKey(accessKey)) {
+      return this.validateStaticKey(accessKey);
+    }
+
+    try {
+      new URL(accessKey);
+    } catch (error) {
+      throw new errors.ServerUrlInvalid(error.message);
+    }
+  }
+
+  private isDynamicAccessKey(accessKey: string) {
+    return !accessKey.startsWith('ss://');
+  }
+
+  private inferAccessKeyName(accessKey: string) {
```

Comment:
Let's keep the default name logic in one place.
I think the simplest for now is to just add an address subtitle for dynamic servers too.

Also, if we think using the address is the better option for the default name, we should do the same for static servers.

Reply:
Yeah, this would result in more consistent behavior, localization-wise. Right now as soon as you edit the default name it is no longer localizable. using the domain would keep that consistent

Reply:
I put all the default name logic in the Server itself - I'm not sure if that's what you meant. Take a look.